mesonbuild/mlog.py

```diff
@@ -103,6 +103,13 @@ def log(*args, **kwargs):
     force_print(*arr, **kwargs)
 
 def warning(*args, **kwargs):
+    from . import environment
+
+    if kwargs.get('location'):
```

Comment:
This should be `if 'location' in kwargs`.

Reply:
The default is `None`...

Reply:
Yeah, I think that is sematically identical because the default default is `None`.

Reply:
However, this line was still wrong :) 

See PR #3090